packages/database/src/prisma-api.ts

```diff
@@ -541,5 +546,40 @@ export function getSdk(
         if (validForProductId) return {defaultMerchantCoupon, defaultCoupon}
       }
     },
+    async transferPurchasesToNewUser({
+      merchantCustomerId,
+      userId,
+    }: {
+      merchantCustomerId: string
+      userId: string
+    }) {
+      const chargesToUpdate = await ctx.prisma.merchantCharge.findMany({
+        where: {
+          merchantCustomerId: merchantCustomerId,
+        },
+      })
+
+      const chargeUpdates = ctx.prisma.merchantCharge.updateMany({
+        where: {
+          merchantCustomerId: merchantCustomerId,
+        },
+        data: {
+          userId: userId,
+        },
+      })
+
+      const purchaseUpdates = ctx.prisma.purchase.updateMany({
+        where: {
+          merchantChargeId: {
+            in: chargesToUpdate.map((c) => c.id),
+          },
+        },
+        data: {
+          userId: userId,
+        },
+      })
+
+      return await ctx.prisma.$transaction([chargeUpdates, purchaseUpdates])
+    },
```

Comment:
main logic, would like to test but these would need to be integration?

Reply:
yeah, I would love to have some integration (unit test integrated with database) tests for these kinds of things. The tests we typically write in Rails, for instance, strike a great balance IMO of acting mostly as unit-level tests while integrating the database. This allows you to meaningfully cover business logic without having to jump into e2e mode (which are higher-effort, slower running tests).

Not sure how we'd set up database-integrating tests in this stack, but it'd be worth a spike at some point.

Reply:
I'd make them a separate suite and just use the docker container probably.

This feels like a spot where it makes a lot of sense to invest in a solution that exercises the database for real, but I'd probably keep stripe portion mocked initially